speechbrain/lobes/models/transformer/Conformer.py

```diff
@@ -17,11 +20,48 @@
     MultiheadAttention,
     PositionalwiseFeedForward,
 )
+from speechbrain.lobes.models.transformer.DCT import DCTConfig
 from speechbrain.lobes.models.transformer.hypermixing import HyperMixing
 from speechbrain.nnet.normalization import LayerNorm
 from speechbrain.nnet.activations import Swish
 
 
+@dataclass
```

Comment:
Why do we need this context stuff? It reads like something that will complexity the understanding of the architecture?

Reply:
The context objects hold the state that has to be remembered for streaming inference (i.e. mostly left context). I wasn't sure what would be the best approach here, so I ended up settling on this pure data design that should be easy to inspect and extend.